Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/storage/VaccinationStorage.kt

```diff
@@ -18,75 +18,67 @@ import javax.inject.Singleton
         baseGson.newBuilder().apply {
-            registerTypeAdapterFactory(containerPostProcessor)
             registerTypeAdapterFactory(CwaCovidCertificate.State.typeAdapter)
         }.create()
     }
 
-    suspend fun load(): Set<VaccinatedPersonData> = mutex.withLock {
+    suspend fun load(): Set<StoredVaccinationCertificateData> = mutex.withLock {
         Timber.tag(TAG).d("load()")
+        return gson
+            .fromJson(
+                prefs.getString(PKEY_VACCINATION_CERT, null) ?: return emptySet(),
+                TYPE_TOKEN
+            )
+    }
+
+    suspend fun save(certificates: Set<StoredVaccinationCertificateData>) = mutex.withLock {
+        Timber.tag(TAG).d("save(%s)", certificates.size)
+        prefs.edit(commit = true) {
+            if (certificates.isEmpty()) {
+                remove(PKEY_VACCINATION_CERT)
+            } else {
+                val rawJson = gson.toJson(certificates, TYPE_TOKEN)
+                putString(PKEY_VACCINATION_CERT, rawJson)
+            }
+        }
+    }
+
+    suspend fun loadLegacyData(): Set<VaccinatedPersonData> = mutex.withLock {
+        Timber.tag(TAG).d("loadLegacyData()")
         val persons = prefs.all.mapNotNull { (key, value) ->
             if (!key.startsWith(PKEY_PERSON_PREFIX)) {
                 return@mapNotNull null
             }
             value as String
-            gson.fromJson<VaccinatedPersonData>(value).also { personData ->
-                Timber.tag(TAG).v("Person loaded: %s", personData.identifier)
-                requireNotNull(personData.identifier)
+            gson.fromJson<VaccinatedPersonData>(value).also { _ ->
+                Timber.tag(TAG).v("Person loaded: %s", key)
             }
         }
-        return persons.toSet().groupDataByIdentifier()
+        return persons.toSet()
     }
 
-    suspend fun save(persons: Set<VaccinatedPersonData>) = mutex.withLock {
-        Timber.tag(TAG).d("save(%s)", persons.size)
-
+    suspend fun clearLegacyData() = mutex.withLock {
+        Timber.tag(TAG).d("clearLegacyData()")
         prefs.edit(commit = true) {
             prefs.all.keys.filter { it.startsWith(PKEY_PERSON_PREFIX) }.forEach {
                 Timber.tag(TAG).v("Removing data for %s", it)
                 remove(it)
             }
-            persons.forEach {
-                if (it.vaccinations.isNotEmpty()) {
-                    val raw = gson.toJson(it)
-                    val identifier = it.identifier
-                    Timber.tag(TAG).v("Storing vaccinatedPerson %s -> %s", identifier, it.vaccinations.size)
-                    putString("$PKEY_PERSON_PREFIX${identifier.groupingKey}", raw)
-                    // TODO: migration should be implemented in (EXPOSUREAPP-11724)
-                }
-            }
         }
     }
 
     companion object {
         private const val TAG = "VaccinationStorage"
         private const val PKEY_PERSON_PREFIX = "vaccination.person."
```

Comment:
Can you add comments to all legacy parts  so they easily identified?

Reply:
updated